pkg/context/context.go

```diff
@@ -1,9 +1,11 @@
-package build
+package context
```

Comment:
I had to do this initially since there was a circular dependency where the `build` package imported the `combustion` package (in order to call `combustion.Configure`) and the `combustion` package imported the `build` package in order to use the `Context` type.

The first dependency was dropped by introducing the `configureCombustion` function type but I hesitated from moving back the `Context` since we're not sure how `build` will turn out in the next couple of iterations as well as `combustion` importing `build` didn't seem right.

We need to think about this further.

I don't have any hard opinions on the naming but it's probably a good idea to move out `CleanUpBuildDir` from this package. I made sure to drop it and all `NewContext()` calls from tests.